ui/src/turing/components/form/standard_ensembler/LinkedRoutesTable.js

```diff
@@ -0,0 +1,166 @@
+  const [pageIndex, setPageIndex] = useState(0);
+  const [allExperiments, setAllExperiments] = useState([]);
+  const [routeToExperimentMappings, setRouteToExperimentMappings] = useState(routes.reduce((m, r) => {m[r.id] = {running: {}, scheduled: {}}; return m}, {}));
+
+  const { start_time, end_time } = useMemo(
+    () => {
+      let current_time = moment.utc();
+      let start_time = current_time.format(appConfig.datetime.format);
+      let end_time = current_time.add(1000, "y").format(appConfig.datetime.format);
+      return { start_time, end_time };
+      },
+    [appConfig]
+  );
+
+  const [{ data: { data: experiments, paging }, isLoaded }] = useXpApi(
+    `/projects/${projectId}/experiments`,
+    {
+      query: {
+        start_time: start_time,
+        end_time: end_time,
+        page: pageIndex + 1,
+        page_size: appConfig.pagination.defaultPageSize,
+        status: "active"
+      },
+    },
+    { data: [], paging: { total: 0 } }
+  );
+
+  const getRouteName = (config, path) => path.split('.').reduce((obj, key) => obj && obj[key], config);
+
+  // this stringified value of routes below allows the React effect below to mimic a deep comparison when changes to the
+  // array routes are made
+  const stringifiedRoutes = JSON.stringify(routes)
+
+  // reset loaded routeToExperimentMappings if treatmentConfigRouteNamePath or routes changes
+  useEffect(() => {
+    if (isAllExperimentsLoaded) {
+      let newRouteToExperimentMappings = routes.reduce((m, r) => {m[r.id] = {running: {}, scheduled: {}}; return m}, {});
+      for (let experiment of allExperiments) {
+        for (let treatment of experiment.treatments) {
+          let configRouteName = getRouteName(treatment.configuration, treatmentConfigRouteNamePath);
+          if (typeof configRouteName === 'string' && configRouteName in newRouteToExperimentMappings) {
+            newRouteToExperimentMappings[configRouteName][getExperimentStatus(experiment).label.toLowerCase()][experiment.id] = experiment;
+          }
+        }
+      }
+      setRouteToExperimentMappings(newRouteToExperimentMappings);
+      setIsButtonPopoverOpen(routes.reduce((m, r) => {m[r.id] = {running: false, scheduled: false}; return m}, {}));
+    }
+  }, [treatmentConfigRouteNamePath, stringifiedRoutes, routes, isAllExperimentsLoaded, allExperiments]);
+
+  useEffect(() => {
+    if (isLoaded) {
+      if (!!experiments && !isAllExperimentsLoaded) {
+        setAllExperiments((curExperiments) => [...curExperiments, ...experiments]);
+      }
+      if (paging.pages > paging.page) {
+        setPageIndex(paging.page);
+      } else {
+        setIsAllExperimentsLoaded(true);
+      }
+    }
+  }, [isLoaded, experiments, paging, isAllExperimentsLoaded]);
+
+  const columns = [
+    {
+      field: "status",
+      width: "5px",
+      render: (_, item) => {
+        const isAssigned = routeToExperimentMappings[item.id] ?
```

Comment:
`status` and `route_name` column computations seem to be duplicated here. Would it make sense to pre-aggregate all column values prior to passing them into the `items` prop of `EuiInMemoryTable`?

Reply:
All the columns (with the exception of `status`) actually use multiple column values (of the rows in `items`, which represents the routes configured) to render the cells of the table. While it's easy to precompute the cells for `status` (`isAssigned` is the only value needed to determine `status`) and store them as a column, `route_name` for example needs both the value of `isAssigned` as well as the `id` of each route. 

This poses a problem because it's not possible to pass 2 (or more) values into the `field` prop of `EuiTableFieldDataColumnType` for the `EuiBasicTable` to extract both values and then pass them as arguments to the function specified in the `render` prop. This means that whatever precomputed values aggregated in advance would only be useful for the partially rendering each cell, because the function of the `render` prop would still have to access or compute other values from each row on-the-fly to determine how the cell ultimately gets rendered.

That said, I think this makes aggregating the computations in advance less useful than expected, since we aren't just extracting a single value to display for each cell, and still relying on the `render` prop to perform the rendering. 

On the other hand, I've not tried precomputing each column as a `ReactNode` component and then getting the table to extract these values as is (without having to rely on the `render` function) i.e. having the `field`-s set as `"column_1"`, `"column_2"`, `"column_3"` ..., and then having these keys within each of the elements in `items` corresponding to `<Column1... />`,  `<Column2... />`,  `<Column3... />`, ... 🤔

Reply:
Just a slight update on this, I think we can just use `id` in the `field` for all of the columns, and then get the `render` function to do something with the `id` since we're almost always using the `routeToExperimentMappings` with the route name.